firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/CRCElements.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ * Chunks.cpp
+ *
+ *  Created on: Nov 3, 2020
+ *      Author: Renji Panicker
+ *
+ * Unit tests to confirm behavior of chunk splitter
```

Comment:
Can you do a comparison against https://github.com/pez-globo/pufferfish-software/blob/develop/backend/tests/unit/protocols/test_crcelements.py and identify what tests and assertions in that file do not have a full equivalent in this file?

Reply:
After comparison following tests and assertions are missing:
- In python there is a separate check_integrity() function to compare computed crc and the one extracted from message.
- there is a get_body() function in python which returns the combined crc and payload which i believe is missing here.

As far as tests i have updated them after our talk, and tried to add similar test cases here. please review